src/aliceVision/voctree/SimpleKmeans.hpp

```diff
@@ -436,9 +462,14 @@ SimpleKmeans<Feature, Distance, FeatureAllocator>::clusterOnce(const std::vector
     assert(checkVectorElements(new_centers, "newcenters init"));
     bool is_stable = true;
 
+    // On small problems enabling multithreading does much more harm than good because thread
+    // creation is relatively expensive.
+    // TODO: Ideally a thread pool would be created before the first iteration and the iterations
+    // would reuse the existing threads.
```

Comment:
Yes I agree.
While a classic kmean can be run on relatively small dataset here it is used for very large datasets to build the vocabulary tree, talking about millions and millions of features.
I'm not against this, I just don't know if the loss on small dataset is significant enough to deal with this.

Reply:
My main motivation is to make tests faster as the tests that run this code were bottlenecks. When running tests in parallel, there's periods of time when there's rather high CPU contention, because many tests want 64 threads for themselves at the same time. Under that scenario this code becomes as much as 1000 times slower.

Long term we should change the code to create a thread pool and reuse the threads. The current situation is that it's problematic to use AliceVision when CPU's are contended. Ideally this shouldn't be the case: one should be able to launch, say, 5 AliceVision pipelines without introducing large processing overheads.